torch/storage.py

```diff
@@ -38,8 +38,41 @@ def size(self) -> int:
         return self.nbytes()
 
     def type(self, dtype: _Optional[str] = None, non_blocking: bool = False) -> T: ...  # type: ignore[empty-body, misc, type-var] # noqa: E704
-    def cuda(self, device=None, non_blocking=False, **kwargs) -> T: ...  # type: ignore[empty-body, misc, type-var] # noqa: E704
-    def hpu(self, device=None, non_blocking=False, **kwargs) -> T: ...  # type: ignore[empty-body, misc, type-var] # noqa: E704
+
+    def cuda(self, device=None, non_blocking=False, **kwargs) -> T:  # type: ignore[type-var] # noqa: E704
+        """Returns a copy of this object in CUDA memory.
+
+        If this object is already in CUDA memory and on the correct device, then
+        no copy is performed and the original object is returned.
+
+        Args:
+            device (int): The destination GPU id. Defaults to the current device.
+            non_blocking (bool): If ``True`` and the source is in pinned memory,
+                the copy will be asynchronous with respect to the host. Otherwise,
+                the argument has no effect.
+            **kwargs: For compatibility, may contain the key ``async`` in place of
+                the ``non_blocking`` argument.
+        """
+        device2 = torch.device('cuda', device) if device else torch.device('cuda')
+        return self.to(device2, non_blocking, **kwargs)
```

Comment:
The "async" argument here has been deprecated for a while, so I think we can take this apportunity to make this PR bc-breaking and remove `**kwargs` here altogether from the public APIs.

Reply:
Yep. Same story - I worked on this PR in the assumption to preserve backward compatibility. If there is a decision to break it, I will make a change.

Reply:
Yes the "not changing it" is definitely the right default behavior for these PR.
This case is a bit different as it makes our new public API significantly worse. So I think it's worth going the slightly more risky route of changing the behavior.
Thanks for being careful in not breaking thing by the way, it is very appreciated.

Reply:
Ok, appreciate if this can be done in a follow up PR. FYI, inside pytorch code only `validate_cuda_device` is called, here:
https://github.com/pytorch/pytorch/blob/3d20a5dd87e9c45e20dbc89210cfc63e7824b2b9/torch/jit/_serialization.py#L181-L182

I don't see calls to `validate_hpu_device`, though both are declared as public APIs and reported to documentation.

Reply:
Ok, I dropped kwargs.